### PR TITLE
Feat: Pre-computing the external model shap graphs before uploading

### PR DESCRIPTION
**Pull request by linus-mt**

### CODE REVIEW

File: sotai/api.py [hunk 1]
```diff
@@ -845,7 +845,7 @@ def post_external_inference(
             ),
             (
                 "files",
-                open( # pylint: disable=consider-using-with
+                open(  # pylint: disable=consider-using-with
```
Comment (willbakst) [inline sotai/api.py:848]: nit: just do a block disable for the whole files argument

File: sotai/utils/shap_utils.py [hunk 2]
```diff
@@ -0,0 +1,388 @@
+"""Shap utils
```
Comment (willbakst) [inline sotai/utils/shap_utils.py:1]: nit: maybe include something here like "Note that this code is based on code from the SHAP package, so we are including the license below".

Also make sure it doesn't overflow on a single line, so maybe best to put this on a new line after Shap utils.

File: sotai/api.py (hunk 1 above)
Comment (linus-mt) [reply sotai/api.py]: Done.

File: sotai/utils/shap_utils.py (hunk 2 above)
Comment (linus-mt) [reply sotai/utils/shap_utils.py]: Done.

File: sotai/api.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+# pylint: disable=consider-using-with
```
Comment (willbakst) [inline sotai/api.py:1]: nit: this disables it for the whole file. Better practice is to do a block disable just for the section, e.g.
```
# pylint: disable=consider-using-with
{code block to disable for}
# pylint: enable=consider-using-with
```

File: requirements.txt [hunk 4]
```diff
@@ -18,3 +18,6 @@ torchvision>=0.15.1
 tox==4.4.8
 tqdm==4.65.0
 twine==4.0.2
+scipy==1.9.1
+shap==0.41.0
```
Comment (willbakst) [inline requirements.txt:22]: are shap and matplotlib requirements for this package? If not, I would remove them.

Also, it looks like scipy is a requirement, so please add this to the pyproject.toml file as well as a dependency.

File: sotai/api.py (hunk 3 above)
Comment (linus-mt) [reply sotai/api.py]: Done.

File: requirements.txt (hunk 4 above)
Comment (linus-mt) [reply requirements.txt:22]: Yeah they are with the new shap related stuff since we re-use the existing shap types and plotting utils.

Comment (willbakst) [reply requirements.txt:22]: Got it I missed that. Make sure then to add all three of these packages to the pyproject.toml dependency list.

**Review by willbakst — APPROVED** [review]

**Pull request merged**